src/dftbp/common/memman.F90

```diff
@@ -30,4 +80,59 @@ pure function incrmntOfArray(currentSize)
 
   end function incrmntOfArray
 
+
+  !> Allocate an array aligned to the n byte boundary (typically used for aligning to 64 bytes)
+  subroutine TAlignedArray_allocateAligned(this, size, alignment)
+
+    !> Instance of the aligned array type
+    class(TAlignedArray), intent(inout) :: this
+
+    !> Lower and upper bound for alignment
+    integer, intent(in) :: size
+
+    !> Alignment (default: 64 bytes)
+    integer, intent(in), optional :: alignment
+
+    if (this%isAllocated) then
+      call error("Aligned array is already allocated")
+    end if
+
+    @:ASSERT(size > 0)
+
+    if (present(alignment)) then
+      @:ASSERT(alignment >= dp)
```

Comment:
You can not use `dp` as the number of bytes used to represent a given type, as it can have any compiler dependent positive value. (Many compilers choose 8 for doubles, but NAG uses 2 for example...). You could use `c_sizeof` instead.